### PR TITLE
Fix/5373  rtl gallery

### LINKED ISSUE #5373: Bug: Gallery not working on rtl
## Description and expectations

Gallery not working on rtl

### Version information

- Tracim version: v4.1.1_build_068

### Screenshots
Capture d’écran de 2022-02-09 13-26-05

________________________________________________

### PR DESCRIPTION
**Pull request by grignards**
Fixes #5373.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link): rtl is part of language selection and is not automatically tested.
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_app_gallery/src/component/GallerySlider.jsx [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import Slider from 'react-slick'
 
+// NOTE 2022-02-16 - SG - Extending Slider
+// to enforce the right initial slide selection at start as 'initialIndex'
+// of react-slick does not work in every case (when several slides are displayed)
```
Comment (raphj) [inline frontend_app_gallery/src/component/GallerySlider.jsx:5]:
```suggestion
// of react-slick does not work in every case (when several slides are displayed)
// use the displayedPictureIndex prop provided by this class instead
```

File: frontend_app_gallery/src/component/Carousel.jsx [hunk 2]
```diff
@@ -15,17 +16,26 @@ export class Carousel extends React.Component {
     this.components = {}
   }
 
-  componentDidUpdate () {
+  componentDidUpdate (prevProps) {
+    // NOTE - 2022-02-17 - SG - avoid changing the position if index are equal
```
Comment (gbonaspetti) [inline frontend_app_gallery/src/component/Carousel.jsx:20]:
```suggestion
    // INFO - SG  - 2022-02-17 - SG - avoid changing the position if index are equal
```

File: frontend_app_gallery/src/component/Carousel.jsx [hunk 3]
```diff
@@ -85,28 +88,32 @@ export class Carousel extends React.Component {
       ]
     }
 
+    // NOTE 2022-02-16 - SG - The support of RTL in react-slick is very buggy
```
Comment (gbonaspetti) [inline frontend_app_gallery/src/component/Carousel.jsx:91]:
```suggestion
    // INFO - SG - 2022-02-16 - The support of RTL in react-slick is very buggy
```

File: frontend_app_gallery/src/component/GallerySlider.jsx [hunk 4]
```diff
@@ -1,5 +1,9 @@
 import Slider from 'react-slick'
 
+// NOTE 2022-02-16 - SG - Extending Slider
```
Comment (gbonaspetti) [inline frontend_app_gallery/src/component/GallerySlider.jsx:3]:
```suggestion
// INFO - SG - 2022-02-16 - Extending Slider
```

File: frontend_app_gallery/src/css/index.styl [hunk 5]
```diff
@@ -61,11 +61,13 @@
     display block
     top 0
     z-index 1
-    & i
+    & i.fa
       position absolute
       top calc(50% - 20px)
       left 50%
-      transform translateX(-50%)
+      // NOTE 2022-02-16 - SG - forcing the scale to (1, 1)
```
Comment (gbonaspetti) [inline frontend_app_gallery/src/css/index.styl:68]:
```suggestion
      // INFO - SG - 2022-02-16 - forcing the scale to (1, 1)
```

File: frontend_app_gallery/src/css/index.styl [hunk 6]
```diff
@@ -164,6 +166,9 @@
 .slick-slide > div
   height 100%
 
+[dir="rtl"] .slick-slide
```
Comment (gbonaspetti) [inline frontend_app_gallery/src/css/index.styl:169]:
```suggestion
[dir='rtl'] .slick-slide
```

Comment (grignards) [inline frontend_app_gallery/src/css/index.styl:169]:
```suggestion
// NOTE 2022-02-17 - SG - disable react-slick rtl handling as it is buggy.
[dir="rtl"] .slick-slide
```

File: frontend_app_gallery/src/component/Carousel.jsx [hunk 7]
```diff
@@ -85,28 +88,32 @@ export class Carousel extends React.Component {
       ]
     }
 
+    // INFO - SG - 2022-02-16 - The support of RTL in react-slick is very buggy
+    // so do not use it and ensure a reverse of the slides list.
+    // Also adaptat props.displayedPictureIndex to be invariant to direction changes outside this component.
```
Comment (raphj) [inline frontend_app_gallery/src/component/Carousel.jsx:93]: typo here @grignards (adaptat)

**Review by raphj — APPROVED** [review]

File: frontend_app_gallery/src/component/Carousel.jsx (hunk 7 above)
Comment (grignards) [reply frontend_app_gallery/src/component/Carousel.jsx]: Corrected.

**Review by raphj — APPROVED** [review]

**Comment by inkhey** [comment]
tested on branch

**Pull request merged**